src/components/nano/NanoContractHistory.js

```diff
@@ -0,0 +1,238 @@
+        nanoHistory = helpers.updateListWs(nanoHistory, tx, NANO_CONTRACT_TX_HISTORY_COUNT);
+
+        // Now update the history
+        setHistory(nanoHistory);
+        setHasAfter(willHaveAfter);
+      }
+    }
+  }, [history, hasAfter, hasBefore, ncId]);
+
+  /**
+   * useCallback is needed here because this method is used as a dependency in the useEffect
+   *
+   * after {string | null} Hash to use for pagination when user clicks to fetch the next page
+   * before {string | null} Hash to use for pagination when user clicks to fetch the previous page
+   */
+  const loadData = useCallback(async (after, before) => {
+    try {
+      const data = await nanoApi.getHistory(ncId, NANO_CONTRACT_TX_HISTORY_COUNT, after, before);
+      if (before) {
+        // When we are querying the previous set of transactions
+        // the API return the oldest first, so we need to revert the history
+        reverse(data.history);
+      }
+      setHistory(data.history);
+
+      if (!after && !before) {
+        // This is the first load without query params, so if has_more === true
+        // we must enable next button
+        setHasAfter(data.has_more);
+        setHasBefore(false);
+        return;
+      }
+
+      if (after) {
+        // We clicked the next button, so we have before page
+        // and we will have the next page if has_more === true
+        setHasAfter(data.has_more);
+        setHasBefore(true);
+        return;
+      }
+
+      if (before) {
+        // We clicked the previous button, so we have next page
+        // and we will have the previous page if has_more === true
+        setHasAfter(true);
+        setHasBefore(data.has_more);
+        if (!data.has_more) {
+          // We are in the first page and clicked the Previous button
+          // so we must clear the query params
+          pagination.clearOptionalQueryParams();
+        }
+        return;
+      }
+    } catch (e) {
+      // Error in request
+      setErrorMessage('Error getting nano contract history.');
+    } finally {
+      setLoading(false);
+    }
+  }, [ncId, pagination]);
+
+  useEffect(() => {
+    // Handle load history depending on the query params in the URL
+    const queryParams = pagination.obtainQueryParams();
+    let after = null;
+    let before = null;
+    if (queryParams.hash) {
+      if (queryParams.page === 'previous') {
+        before = queryParams.hash;
+      } else if (queryParams.page === 'next') {
+        after = queryParams.hash;
+      } else {
+        // Params are wrong
+        pagination.clearOptionalQueryParams();
+      }
+    }
+
+    loadData(after, before);
+
+  }, [location, loadData, pagination]);
+
+  useEffect(() => {
+    // Handle new txs in the network to update the list in real time
+    WebSocketHandler.on('network', handleWebsocket);
+
+    return () => {
+      WebSocketHandler.removeListener('network', handleWebsocket);
+    };
+  });
```

Comment:
question: Isn't this missing a `[]` parameter? I would expect the event handler to be initialized only once, and not on every render.
```suggestion
  }, []);
```

Reply:
Yes, it's supposed to run only once, thanks https://github.com/HathorNetwork/hathor-explorer/pull/292/commits/526d76871f98e8df03f6f3ab9407c791b5de8792

Reply:
Actually after adding the empty array as dependency, the linter complained that I should've added the method `handleWebsocket` as a dependency, because this `useEffect` uses it. With this method as dependency, I had to change it to a `useCallback`, so I had to do one small refactor: https://github.com/HathorNetwork/hathor-explorer/pull/292/commits/960d86bc98537907c965164885b27a7dc125790b